Main.bb

```diff
@@ -1960,6 +1962,29 @@ Function CreateDoor.Doors(lvl, x#, y#, z#, angle#, room.Rooms, dopen% = False,
 	
 	d\MTFClose = True
 	
+	Local d2.Doors
+	If useCollisionMesh Then
+		For d2.Doors = Each Doors
+			If d2 <> d Then
+				If d2\DoorHitOBJ <> 0 Then
+					d\DoorHitOBJ = CopyEntity(d2\DoorHitOBJ,d\frameobj)
+					EntityAlpha d\DoorHitOBJ,0.0
+					EntityFX d\DoorHitOBJ,1
+					EntityType d\DoorHitOBJ,HIT_MAP
+					EntityColor d\DoorHitOBJ,255,0,0
+					Exit
+				EndIf
+			EndIf
+		Next
+		If d\DoorHitOBJ=0 Then
+			d\DoorHitOBJ = LoadMesh_Strict("GFX\doorhit.b3d",d\frameobj)
```

Comment:
Is there a need to use a mesh instead of a simple AABB box? Or any sort of collider for that matter, couldn't the character just be moved out from the doorway when it's closing? Also, won't this leak memory because the HitObjs are never freed?

Reply:
Well, the character getting moved away is sort of a workaround which is already supposed to be done by the doors but doesn't work correctly.
It's only used for the doors that are pretty much required to have this system (so that the character can't glitch through it to cause the sequence to be broken or such).
I don't actually know how to use EnityBox properly, and I use the HIT_MAP type which is programmed to be sphere to polygon collisions, not sphere to box (that would require to initialize yet another type of collision, one that uses polygon to box as the game doesn't has that yet, only sphere to sphere and sphere to box and I personally don't know how much that would have an impact on the game's performance).
I recently just saw that there could be a potential memory leak, but only when the door is supposed to be removed using RemoveDoor, as ClearWorld in NullGame already takes care of that (but I can still add that nonetheless, just to be sure).